core/domain/state_domain.py

```diff
@@ -941,10 +940,6 @@ def validate(self, expected_content_id_list):
                 raise utils.ValidationError(
                     'Expected content_id to be a string, received %s'
                     % content_id)
-            if not isinstance(language_code_to_written_translation, dict):
```

Comment:
When you delete lines, can you leave a comment saying why you did so?
So, it doesn't take a round of review to make it clear to the reviewers. Thanks.

Reply:
This and the next change was a mistake. Reverted it and added tests for it